pretests/docker_test_images/docker_test_images.py

```diff
@@ -0,0 +1,76 @@
+r"""
+Summary
+---------
+
+Pre-testing pulling and listing of test image(s) into sysinfo results
+
+Operational Summary
+----------------------
+
+#. Parse the default test image into FQIN format
+#. Pull the default, and any configured ``extra_fqins_csv`` images
+#. Log a listing of all current images to debug and a sysinfo file
+#. Optionally, update ``config_defaults/defaults.ini`` (if it exists)
+   to preserve all pulled images.  Configured by ``update_defaults_ini``
+   option (default: False)
+
+Prerequisites
+---------------
+
+The default image and any ``extra_fqins_csv`` images exist and are pull-able
+within the testing timeout period.  If it is to be updated with pulled images,
+the file ``config_defaults/defaults.ini`` must exist.
+"""
+
+import os.path
+from dockertest.subtest import Subtest
+from dockertest.images import DockerImages
+from dockertest.dockercmd import DockerCmd
+from dockertest.config import get_as_list
+from dockertest.config import CONFIGCUSTOMS
+from dockertest.config import ConfigSection
+
+
+class docker_test_images(Subtest):
+    """Pull, then log a listing of docker images present on system"""
+
+    def initialize(self):
+        super(docker_test_images, self).initialize()
+        self.stuff['di'] = di = DockerImages(self)
+        extra_fqins_csv = self.config.get('extra_fqins_csv', '')
+        update_defaults_ini = self.config.get('update_defaults_ini', False)
+        self.stuff['defaults_ini'] = os.path.join(CONFIGCUSTOMS,
+                                                  'defaults.ini')
+        defaults_ini_exists = os.path.isfile(self.stuff['defaults_ini'])
+        self.stuff['update'] = update_defaults_ini and defaults_ini_exists
+        self.stuff['fqins'] = [di.default_image] + get_as_list(extra_fqins_csv)
+
+    def run_once(self):
+        super(docker_test_images, self).run_once()
+        for fqin in self.stuff['fqins']:
+            # TODO: Support pulling/verifying with atomic command
+            DockerCmd(self, 'pull', [fqin]).execute()
```

Comment:
You're right, it is. I think the issue is that autotest-local (on the command line) doesn't log DEBUG by default. Anyway, the symptom is: I run `autotest-local run docker` and it hangs for many minutes in the pull. Maybe a `loginfo()` would fix it.